### ISSUE TITLE
SSR crashes

### ISSUE THREAD

**Issue by Schroedingers-Cat** [chohner/m4l-SSR-Remote#1]
I'm using the m4l-SSR-Remote within Ableton Live 9.7.4 and Max 7 to control two instances of the SSR 0.4.2 on Ubuntu 16.04. It works for some seconds, but both instances crash with the following error message:

> terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::bad_weak_ptr> >'
  what():  tr1::bad_weak_ptr
Abgebrochen (Speicherabzug geschrieben)

What could cause this issue? I copied the modified files from this repo to `C:\Program Files\Cycling '74\Max 7\resources\packages\max-mxj\java-classes\classes\net\tcp`.

**Comment by chohner** [comment]
Hi! Does it work with a single instance? If you are comfortable with compiling SSR yourself, could you try the version straight from GitHub? I think Boost was completely dropped.

**Comment by Schroedingers-Cat** [comment]
It also crashes a single instance of SSR. I just compiled the lastest version from github/master and it still crashes. Error message is a bit shorter:
> terminate called after throwing an instance of 'std::bad_weak_ptr'
  what():  bad_weak_ptr
Abgebrochen (Speicherabzug geschrieben)

Any idea?

**Comment by chohner** [comment]
Hm no, I'm sorry. I'm also currently able to really test this.
If you compiled your own version of SSR, you can start it with `--end-of-message-character=10`. Then you won't the modified tcp files (that only hack in a binary `0` as a line-ending)

**Comment by Schroedingers-Cat** [comment]
Yes, I forgot to mention that I did start the compiled version with that parameter and placed the origincal tcp java classes back into the max folder.

Could this rather be an issue with the SSR itself?

**Comment by chohner** [comment]
I think something in the network interface is crashing SSR, yes. It's hard to say exactly - something is deleting an object that a `std::weak_ptr` references but which should be accessed.

Does this happen also at slow update rates?

**Comment by Schroedingers-Cat** [comment]
Had the update interval set to 800. Both instances crashed after roughly an hour.

**Comment by Schroedingers-Cat** [comment]
Update: When running two instances, the second instance crashes after 10 minutes and the first instance crashes after an hour. Happened three times in a row now.

**Comment by chohner** [comment]
Interesting. I can try to look into this for the weekend but afaik the network interface on SSR side of things is not the most stable thing in the world.
